frontend-react/src/components/header/DAPHeader.test.tsx

```diff
@@ -0,0 +1,70 @@
+import routeData from "react-router";
+
+import { useDAP } from "./DAPHeader";
+
+describe("Test useDAP for DAPHeader component", () => {
+    const useLocation = jest.spyOn(routeData, "useLocation");
+
+    test("Check DAP for root homepage, login and TOS pages (public)", () => {
+        useLocation.mockReturnValue({ pathname: "/" } as any);
+        expect(useDAP("production")).toBe(true);
+        expect(useDAP("development")).toBe(false);
+        expect(useDAP(undefined)).toBe(false);
```

Comment:
what happens when this env var is a blank `string` or `null`? can we write a test for that?

Reply:
I think we're okay will null because it's typed as `string | undefined`; null should raise a compile time error. But an empty string should definitely be tested, and accounted for in the hook. Maybe with a `console.error`?